src/tasks/harvest_event_worker.js

```diff
@@ -33,68 +72,70 @@ function start () {
-
-                event.update({
-                  name: item.name,
-                  platform: 'meetup',
-                  platform_identifier: `${item.id}`,
-                  description: htmlToText.fromString(item.description),
-                  location: location,
-                  rsvp_count: item.yes_rsvp_count,
-                  url: item.link,
-                  group_id: item.group.id,
-                  group_name: item.group.name,
-                  group_url: `https://www.meetup.com/${item.group.urlname}`,
-                  formatted_time: startTime.tz('Asia/Singapore').format('DD MMM YYYY, ddd, h:mm a'),
-                  start_time: startTime.toDate(),
-                  end_time: endTime.toDate(),
-                  latitude: (item.venue ? item.venue.lat : null),
-                  longitude: (item.venue ? item.venue.lon : null)
-                }).then(() => {
-                  console.log('Updated the record for', item.name)
-                })
-              })
-          })
-          done()
+      const eventResponses = await harvester.fetchGroupEvents(job.data)
+      const allGroupEvents = eventResponses.events
+      console.log('=====================================================')
+      console.log(`Harvested ${allGroupEvents.length} events from ${job.data.urlname}`)
+
+      for (const item of allGroupEvents) {
+        console.log('Event:', item.name)
+
+        const [event, created] = await db.Event.findOrBuild({
+          where: {
+            platform: 'meetup',
+            platform_identifier: `${item.id}`
+          }
         })
-        .catch(error => {
-          console.error(error.message)
-          Sentry.captureException(err)
-          throw error
+
+        let location = ''
+        if (item.venue) {
+          location = item.venue.name
+
+          if (item.venue.address_1) {
+            location += `, ${item.venue.address_1}`
+          }
+        }
+
+        const startTime = moment(`${item.local_date} ${item.local_time} +08:00`, 'YYYY-MM-DD HH:mm Z')
+        const endTime = moment(startTime).add(item.duration, 'milliseconds')
+
+        await event.update({
+          name: item.name,
+          platform: 'meetup',
+          platform_identifier: `${item.id}`,
+          description: htmlToText.fromString(item.description),
+          location: location,
+          rsvp_count: item.yes_rsvp_count,
+          url: item.link,
+          group_id: item.group.id,
+          group_name: item.group.name,
+          group_url: `https://www.meetup.com/${item.group.urlname}`,
+          formatted_time: startTime.tz('Asia/Singapore').format('DD MMM YYYY, ddd, h:mm a'),
+          start_time: startTime.toDate(),
+          end_time: endTime.toDate(),
+          latitude: (item.venue ? item.venue.lat : null),
+          longitude: (item.venue ? item.venue.lon : null)
         })
+
+        console.log('Updated the record for', item.name)
+      }
+
+      done()
```

Comment:
Quite a few promises were unhandled here. We can simplify and flatten most of it, using async-await.

I also loop through events in sequence rather than in parallel. It makes the logs easier to read.